tensorflow/src/TensorFlow/Internal/FFI.hs

```diff
@@ -187,8 +188,12 @@ createTensorData t = do
     -- Read data.
     len <- fromIntegral <$> Raw.tensorByteSize t
     bytes <- castPtr <$> Raw.tensorData t :: IO (Ptr Word8)
-    -- TODO(fmayle): Don't copy the data.
-    v <- S.fromList <$> peekArray len bytes
+    -- Note: We would like to avoid the copy by creating an S.Vector directly
+    -- from 'bytes' and calling Raw.deleteTensor when it gets GC'd, but we can't
+    -- because the vector may outlive the tensorflow session and become invalid.
```

Comment:
Right. I had done that and was getting an error on process exit for the benchmark, which led me to think that the ffi tensors were being free'd on session deletion or something. I've looked into it more now and realize that they aren't associated with the session, so that wasn't the issue. I can't reproduce the error anymore... so, going with the implementation that doesn't copy at all.